src/typedefs.jl

```diff
@@ -21,13 +21,14 @@ reusing operations that only work on arrays without any null values.
 immutable NullableArray{T, N} <: AbstractArray{Nullable{T}, N}
     values::Array{T, N}
     isnull::Array{Bool, N}
+    ref::Vector{UInt8}
```

Comment:
Maybe find a different name: we'll likely have to add the same hack to CategoricalArrays, which have a `refs` field. `parent`, `memblock`, `memregion`, `buffer`?